packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -52,6 +52,8 @@ class DebuggerController extends DisposableController
       isPaused: isPaused,
     );
 
+    setGlobal(EvalService, evalService);
```

Comment:
Creating a controller object should not have side effects like setting a global. Either move this logic to other places globals are set or just hang the evalService off the DebuggerController object. All places the Console was used can just be given access to the DebuggerController. Generally I think that is idiomatic as the DebugController is the source of truth for things like whether you are paused at a breakpoint and what the selected stack frame is so there is no reason the console should not know those things.

Reply:
Moved state that needs to be shared between inspector, debugger, and console, from DebuggerController to AppState owned by ConnectedApp.